ProjectShop/ProductApp/views.py

```diff
@@ -55,6 +61,35 @@ class CategoriesView(generic.ListView):
     paginate_by = 12
 
 
+class CategoryDetailView(CategoryListMixin, generic.DetailView):
+    model = ProductCategory
+    context_object_name = 'category_detail'
+    template_name = 'ProductApp/category_detail.html'
+
+    def get_queryset(self):
+        return ProductCategory.objects.all()
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        products = self.get_related_products()
+        context['products'] = products
+        context['page_obj'] = products
+        return context
+
+    def get_related_products(self):
+        queryset = Product.objects.filter(categories=self.object)
+        ordered_queryset = queryset.order_by('-stock_quantity')
+        paginator = Paginator(ordered_queryset, 12)
+        try:
+            page = self.request.GET.get('page')
```

Comment:
If I'm not mistaken - we need to use double quotes everywhere.

Reply:
Thank you. I appreciate that. The changes have been made.